import_flow_events.py

```diff
@@ -89,23 +89,31 @@
       -- but redshift doesn't support that.
       flow_time BIGINT NOT NULL ENCODE lzo,
       flow_id VARCHAR(64) NOT NULL DISTKEY ENCODE lzo,
-      type VARCHAR(64) NOT NULL ENCODE lzo
+      type VARCHAR(64) NOT NULL ENCODE lzo,
+      export_date DATE NOT NULL ENCODE lzo
     );
 """
 
 Q_CHECK_FOR_DAY = """
     SELECT timestamp FROM flow_events
-    WHERE timestamp::DATE = '{day}'::DATE
+    WHERE timestamp::DATE <= '{day}'::DATE + '1 day'::INTERVAL
+      AND timestamp::DATE >= '{day}'::DATE - '1 day'::INTERVAL
+      AND export_date = '{day}'::DATE;
     LIMIT 1;
 """
 
-Q_CLEAR_DAY_METADATA = """
-    DELETE FROM flow_metadata
-    WHERE begin_time::DATE = '{day}'::DATE;
-"""
 Q_CLEAR_DAY_EVENTS = """
     DELETE FROM flow_events
-    WHERE timestamp::DATE = '{day}'::DATE;
+    WHERE timestamp::DATE <= '{day}'::DATE + '1 day'::INTERVAL
+      AND timestamp::DATE >= '{day}'::DATE - '1 day'::INTERVAL
+      AND export_date = '{day}'::DATE;
+"""
+Q_CLEAR_DAY_METADATA = """
+    DELETE flow_metadata
+    FROM flow_metadata AS m
+    LEFT JOIN flow_events AS e
+      ON m.flow_id = e.flow_id
+    WHERE e.flow_id IS NULL;
```

Comment:
> Not sure how much difference it will make in practice though.

Maybe I did it wrong but this actually made it even slower when I tried it just now. I've reinstated the script from `master` and kicked off a fresh import. Just fyi in case you happen to be looking at flow data during the course of your Monday.